include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
-                        dataSelection,
-                        (meshesPathName + dataLabelsDetectorFrequency(0)).c_str(),
-                        detectorFrequencies.data());
+                            // write actual data
+                            mesh_n[dir].storeChunk(::openPMD::shareRaw(tmpBuffer), offset_n, extent_n);
+                            openPMDdataFile.flush();
+                        }
+                    }
 
-                    /* save SI unit as attribute together with data set */
+                    // end: write observer
+
+
+                    // start: write frequencies
+                    // prepare omega mesh
+                    ::openPMD::Mesh mesh_omega = openPMDdataFileIteration.meshes[dataLabelsDetectorFrequency(-1)];
+                    // write mesh attributes
+                    mesh_omega.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+                    mesh_omega.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    mesh_omega.setGridSpacing(std::vector<double>{1.0, 1.0, 1.0});
+                    mesh_omega.setGridGlobalOffset(std::vector<double>{0.0, 0.0, 0.0});
+                    mesh_omega.setGridUnitSI(1.0);
+                    mesh_omega.setAxisLabels(std::vector<std::string>{"None", "detector frequency index", "None"});
+                    std::map<::openPMD::UnitDimension, double> const unitDimensions_omega{
+                        {::openPMD::UnitDimension::T, -1.0}};
+                    mesh_omega.setUnitDimension(unitDimensions_omega);
+
+                    // write mesh attributes
+                    ::openPMD::MeshRecordComponent omega_mrc = mesh_omega[dataLabelsDetectorFrequency(0)];
                     const picongpu::float_64 factorOmega = 1.0 / UNIT_TIME;
-                    hdf5DataFile.writeAttribute(
-                        currentStep,
-                        radSplashType,
-                        (meshesPathName + dataLabelsDetectorFrequency(0)).c_str(),
-                        "unitSI",
-                        &factorOmega);
-
-                    /* position */
-                    std::vector<float_X> positionMesh(simDim, 0.0); /* there is no offset - zero */
-                    hdf5DataFile.writeAttribute(
-                        currentStep,
-                        splashFloatXType,
-                        (meshesPathName + dataLabelsDetectorFrequency(0)).c_str(),
-                        "position",
-                        1u,
-                        splash::Dimensions(simDim, 0, 0),
-                        &(*positionMesh.begin()));
+                    omega_mrc.setUnitSI(factorOmega);
+                    omega_mrc.setPosition(std::vector<double>{0.0, 0.0, 0.0});
 
+                    // reshape abstract MeshRecordComponent
+                    ::openPMD::Datatype datatype_omega
+                        = ::openPMD::determineDatatype(::openPMD::shareRaw(detectorFrequencies.data()));
```

Comment:
```suggestion
                        = ::openPMD::determineDatatype<decltype(detectorFrequencies)::value_type>();
```
does not use a dummy `shareRaw`

Reply:
There seems to be a typo in here

Reply:
(Or the CI failed in some other point and this line only caused an error. 😕 )